### PR TITLE
app-admin/etcd-wrapper: remove slotting and complexity

### PR DESCRIPTION
**Pull request by steveeJ**
Passing arguments via this environment variable is more convenient than e.g.
overriding the ExecStart line in a drop-in.

### CODE REVIEW

**Comment by crawford** [comment]
I think we use _OPTS elsewhere. We should probably standardize on one name.

**Comment by steveeJ** [comment]
@crawford +1 on standardizing, you know more about the projects than I do so just tell me what it's going to be and I'll adapt the PR.

**Comment by crawford** [comment]
Let's go with ETCD_OPTS

**Comment by crawford** [comment]
LGTM

**Comment by steveeJ** [comment]
@crawford I reworked this one and checked it on a new build. After logging in per SSH, one needs to do the following to be able to use it:

1. write `ETCD_TAG=v2.3.7` to /var/lib/coreos/versions/etcd
2. `sudo systemctl start etcd-wrapper`

`ETCD_TAG` could also be provided as a dropin `Environment=` for the service file. I think this is a good balance of convenience and responsibility for the user.

/cc @philips

**Comment by crawford** [comment]
> 1. write ETCD_TAG=v2.3.7 to /var/lib/coreos/versions/etcd

I don't like this. Why even provide the option? I think the `Environment=` is the correct mechanism.

File: app-admin/etcd-wrapper/etcd-wrapper-0.0.1-r0.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+SLOT=0
+
+DEPEND=""
+RDEPEND=">=app-emulation/rkt-1.9.1[rkt_stage1_fly]"
+
+S=${WORKDIR}
+
+src_install() {
+    dobin "${FILESDIR}"/etcd-wrapper
+  	systemd_dounit "${FILESDIR}"/etcd-wrapper.service
```
Comment (mischief) [inline app-admin/etcd-wrapper/etcd-wrapper-0.0.1-r0.ebuild]: this indentation seems strange, at least in github ui.

**Comment by steveeJ** [comment]
@crawford 

> I don't like this. Why even provide the option? I think the Environment= is the correct mechanism.

You can do either, or both even if you want. The `EnvironmentFile` path starts with a `-` declaring it non-critical. You can easily just use a drop-in that containts `Environment=ETCD_TAG=v2.3.7`.

File: app-admin/etcd-wrapper/etcd-wrapper-0.0.1-r0.ebuild (hunk 1 above)
Comment (steveeJ) [inline app-admin/etcd-wrapper/etcd-wrapper-0.0.1-r0.ebuild]: Fixed, thanks

**Comment by crawford** [comment]
> The EnvironmentFile path...

But why is this here at all? I think we should remove this since we are no longer providing a default.

File: app-admin/etcd-wrapper/files/etcd-wrapper.service [hunk 2]
```diff
@@ -0,0 +1,19 @@
+[Unit]
+Description=etcd (System Application Container)
+Conflicts=etcd.service
+Conflicts=etcd2.service
+
+[Service]
+Type=notify
```
Comment (crawford) [inline app-admin/etcd-wrapper/files/etcd-wrapper.service]: Let's add an empty `Environment=ETCD_TAG=` so it's obvious that it needs to be overridden.

**Comment by steveeJ** [comment]
> But why is this here at all? I think we should remove this since we are no longer providing a default.

It's so people don't have to necessarily create a drop-in. Placing an environment file is easier as it doesn't require a `daemon-reload`.

File: app-admin/etcd-wrapper/files/etcd-wrapper.service (hunk 2 above)
Comment (steveeJ) [inline app-admin/etcd-wrapper/files/etcd-wrapper.service]: ACK

**Comment by crawford** [comment]
> Placing an environment file is easier as it doesn't require a daemon-reload.

Using EnvironmentFile is frowned upon in the systemd community. Having multiple ways to configure this is just going to confuse people. Please remove.

**Comment by steveeJ** [comment]
> Using EnvironmentFile is frowned upon in the systemd community. Having multiple ways to configure this is just going to confuse people. Please remove.

Frankly, I don't think that the mere fact that something is frowned upon is a valid argument nor is it an explanation at all. I don't see why it would be bad to use that option, but I'll remove the line for the sake of moving forward.
Sorry folks, I've tried to save your systemd from an unnecessary reload!

**Comment by crawford** [comment]
LGTM

**Pull request merged**